cmd/restic/cmd_check.go

```diff
@@ -263,10 +311,28 @@ func runCheck(opts CheckOptions, gopts GlobalOptions, args []string) error {
 
 	switch {
 	case opts.ReadData:
-		doReadData(1, 1)
+		Verbosef("read all data\n")
+		doReadData(doSelectPacksByNumbers(1, 1))
 	case opts.ReadDataSubset != "":
-		dataSubset, _ := stringToIntSlice(opts.ReadDataSubset)
-		doReadData(dataSubset[0], dataSubset[1])
+		var packs *restic.IDSet
+		dataSubset, err := stringToIntSlice(opts.ReadDataSubset)
+		if err == nil {
+			bucket := dataSubset[0]
+			totalBuckets := dataSubset[1]
+			packs = doSelectPacksByNumbers(bucket, totalBuckets)
+			packCount := uint64(len(*packs))
+			Verbosef("read group #%d of %d data packs (out of total %d packs in %d groups)\n", bucket, packCount, chkr.CountPacks(), totalBuckets)
+		} else {
+			percentage, err := parsePercentage(opts.ReadDataSubset)
+			if err == nil {
```

Comment:
This is the expected case, so it should be unindented. Just store the error in _, or panic if it is non-nil.

Reply:
Fixed in https://github.com/restic/restic/pull/3038/commits/4943b1ebda8ad41c5cd485e202336e780a1e8c77